app/controllers/registrations_controller.rb

```diff
@@ -52,13 +52,13 @@ def create
     else
       existing_user = User.where_case_insensitive('email', sign_up_params[:email]).first
       if existing_user.present?
-        if existing_user.accept_terms?
+        if existing_user.invitation_token.present? && !existing_user.accept_terms?
```

Comment:
not sure why was
```
existing_user.accept_terms?
```
instead of
```
existing_user.accept_terms.nil?
```
anyway adding the invitation_token makes it safer. Thanks